lib/utilities/attrValidators.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * More validators for class attributes. 
+ * @class AttrValidators
+ * @static
+ */
+var AttrValidators = {
+
+  /**
+   * Returns a function that checks if a value is an instance of the given
+   * contructor isInstanceOf.
+   *
+   * @method isInstanceOf
+   * @param ctor {Object} The contructor function
+   */
+  isInstanceOf: function(ctor) {
+    if (!ctor) {
+      console.error(ctor);
+      console.error('Constructor object to check against was falsy.');
+      return false;
+    }
+
+    return function (val, attrName) {
+      return (val instanceof ctor);
+    };
+  },
+
+  /**
+   * Returns a function that checks if a value is in the given constants object.
+   *
+   * @method isValueIn
+   * @param obj {Object} The object of constants (key integer pairs)
+   */
+  isValueIn: function (obj) {
+    if (!obj) {
+      console.error(obj);
+      console.error('Constants object to check against was falsy.');
+      return false;
+    }
+    
+    return function (val, attrName) {
+      var values = [];
+
+      for (var key in obj) {
+        values.push(obj[key]);
+      }
+
+      return values.indexOf(val) !== -1;
```

Comment:
It seems like you could have just checked for the presence of `val` in the loop above. Why incur this extra cost? This could be significant for large objects. In fact, even checking in the loop seems expensive. Isn't there a faster lookup for object keys?